ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/JdbcLedgerDaoSpec.scala

```diff
@@ -270,6 +271,140 @@ class JdbcLedgerDaoSpec
+        response <- ledgerDao.storeConfigurationEntry(
+          offset,
+          offset + 1,
+          None,
+          Instant.EPOCH,
+          "config-rejection-0",
+          participantId,
+          proposedConfig,
+          Some("bad config")
+        )
+        storedConfig <- ledgerDao.lookupLedgerConfiguration()
+        entries <- ledgerDao.getConfigurationEntries(offset, offset + 1).runWith(Sink.seq)
+
+      } yield {
+        response shouldEqual PersistenceResponse.Ok
+        startingConfig shouldEqual storedConfig
+        entries shouldEqual List(
+          offset -> ConfigurationEntry
+            .Rejected("config-rejection-0", participantId, "bad config", proposedConfig)
+        )
+      }
+    }
+
+    "refuse to persist invalid configuration entry" in {
+      val offset0 = nextOffset()
+      val participantId = Ref.LedgerString.assertFromString("participant-0")
+      for {
+        config <- ledgerDao.lookupLedgerConfiguration().map(_.getOrElse(defaultConfig))
+
+        // Store a new configuration with a known submission id
+        resp0 <- ledgerDao.storeConfigurationEntry(
+          offset0,
+          offset0 + 1,
+          None,
+          Instant.EPOCH,
+          "refuse-config-0",
+          participantId,
+          config.copy(generation = config.generation + 1),
+          None
+        )
+        newConfig <- ledgerDao.lookupLedgerConfiguration().map(_.get)
+
+        // Submission with duplicate submissionId is rejected
+        offset1 = nextOffset()
+        resp1 <- ledgerDao.storeConfigurationEntry(
+          offset1,
+          offset1 + 1,
+          None,
+          Instant.EPOCH,
+          "refuse-config-0",
+          participantId,
+          newConfig.copy(generation = config.generation + 1),
+          None
+        )
+
+        // Submission with mismatching generation is rejected
+        offset2 = nextOffset()
+        resp2 <- ledgerDao.storeConfigurationEntry(
+          offset2,
+          offset2 + 1,
+          None,
+          Instant.EPOCH,
+          "refuse-config-1",
+          participantId,
+          config,
+          None
+        )
+
+        // Submission with unique submissionId and correct generation is accepted.
+        offset3 = nextOffset()
+        lastConfig = newConfig.copy(generation = newConfig.generation + 1)
+        resp3 <- ledgerDao.storeConfigurationEntry(
+          offset3,
+          offset3 + 1,
+          None,
+          Instant.EPOCH,
+          "refuse-config-2",
+          participantId,
+          lastConfig,
+          None
+        )
+        lastConfigActual <- ledgerDao.lookupLedgerConfiguration().map(_.get)
+
+        entries <- ledgerDao.getConfigurationEntries(offset0, offset3 + 1).runWith(Sink.seq)
+      } yield {
+        resp0 shouldEqual PersistenceResponse.Ok
+        resp1 shouldEqual PersistenceResponse.Duplicate
+        resp2 shouldEqual PersistenceResponse.Duplicate
```

Comment:
A mismatching config generation results in `Duplicate`?

Reply:
Will rewrite to store a rejection reason instead.